sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveQl.scala

```diff
@@ -557,8 +557,11 @@ https://cwiki.apache.org/confluence/display/Hive/Enhanced+Aggregation%2C+Cube%2C
             "TOK_TABLEPROPERTIES"),
           children)
       val (db, tableName) = extractDbNameTableName(tableNameParts)
-
-      CreateTableAsSelect(db, tableName, nodeToPlan(query), allowExisting != None, Some(node))
+      if (ignores.exists(_ != None)) {
+        CreateTableAsSelect(db, tableName, nodeToPlan(query), allowExisting != None, Some(node))
+      } else {
+        CreateTableAsSelect(db, tableName, nodeToPlan(query), allowExisting != None, None)
```

Comment:
When `hive.convertCTAS` is false, it goes to `execution.CreateTableAsSelect` and calls `createTable` in `HiveMetastoreCatalog` to create the table. For now, `createTable` will choose default if the given `CreateTableDesc` is `None`.